Deflake DrainTest

During CI or even with just repeated test runs, DrainTest is failing:

For example:

```
2026-07-10T15:22:48.3542511Z [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.451 s <<< FAILURE! -- in io.quarkus.resteasy.reactive.server.test.stress.DrainTest
2026-07-10T15:22:48.3550059Z [ERROR] io.quarkus.resteasy.reactive.server.test.stress.DrainTest.testSyncHttp1 -- Time elapsed: 1.974 s <<< FAILURE!
2026-07-10T15:22:48.3551588Z java.lang.AssertionError: Request failed
2026-07-10T15:22:48.3552727Z 	at io.quarkus.resteasy.reactive.server.test.stress.DrainTest.runTest(DrainTest.java:99)
2026-07-10T15:22:48.3554282Z 	at io.quarkus.resteasy.reactive.server.test.stress.DrainTest.testSyncHttp1(DrainTest.java:59)
2026-07-10T15:22:48.3556060Z Caused by: java.lang.IllegalStateException: Client is closed
2026-07-10T15:22:48.3557299Z 	at io.vertx.core.net.impl.tcp.NetClientImpl.connectInternal(NetClientImpl.java:228)
2026-07-10T15:22:48.3558782Z 	at io.vertx.core.net.impl.tcp.NetClientImpl.connectInternal(NetClientImpl.java:173)
2026-07-10T15:22:48.3560219Z 	at io.vertx.core.http.impl.tcp.TcpHttpClientTransport.connect(TcpHttpClientTransport.java:178)
2026-07-10T15:22:48.3561734Z 	at io.vertx.core.http.impl.tcp.TcpHttpClientTransport.connect(TcpHttpClientTransport.java:285)
2026-07-10T15:22:48.3565305Z 	at io.vertx.core.http.impl.SharedHttpClientConnectionGroup$Pool.connect(SharedHttpClientConnectionGroup.java:240)
2026-07-10T15:22:48.3578130Z 	at io.vertx.core.internal.pool.SimpleConnectionPool.connect(SimpleConnectionPool.java:251)
2026-07-10T15:22:48.3579641Z 	at io.vertx.core.internal.pool.SimpleConnectionPool$Remove$1.run(SimpleConnectionPool.java:429)
2026-07-10T15:22:48.3580893Z 	at io.vertx.core.internal.pool.Task.runNextTasks(Task.java:43)
2026-07-10T15:22:48.3582037Z 	at io.vertx.core.internal.pool.CombinerExecutor.submit(CombinerExecutor.java:91)
2026-07-10T15:22:48.3583372Z 	at io.vertx.core.internal.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:242)
2026-07-10T15:22:48.3584772Z 	at io.vertx.core.internal.pool.SimpleConnectionPool.remove(SimpleConnectionPool.java:500)
2026-07-10T15:22:48.3586472Z 	at io.vertx.core.internal.pool.SimpleConnectionPool$Slot.onRemove(SimpleConnectionPool.java:136)
2026-07-10T15:22:48.3588225Z 	at io.vertx.core.http.impl.SharedHttpClientConnectionGroup$Pool.lambda$connect$0(SharedHttpClientConnectionGroup.java:244)
2026-07-10T15:22:48.3590099Z 	at io.vertx.core.http.impl.http1.Http1ClientConnection.writeClose(Http1ClientConnection.java:803)
2026-07-10T15:22:48.3599338Z 	at io.vertx.core.net.impl.VertxConnection.terminateClose(VertxConnection.java:238)
2026-07-10T15:22:48.3601066Z 	at io.vertx.core.net.impl.VertxConnection.handleClose(VertxConnection.java:231)
2026-07-10T15:22:48.3607438Z 	at io.vertx.core.net.impl.VertxHandler.close(VertxHandler.java:175)
2026-07-10T15:22:48.3609006Z 	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:635)
2026-07-10T15:22:48.3610638Z 	at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:1015)
2026-07-10T15:22:48.3611792Z 	at io.netty.channel.Channel.close(Channel.java:282)
2026-07-10T15:22:48.3613040Z 	at io.vertx.core.net.impl.VertxConnection.handleShutdown(VertxConnection.java:158)
2026-07-10T15:22:48.3614453Z 	at io.vertx.core.http.impl.http1.Http1Connection.closeInternal(Http1Connection.java:84)
2026-07-10T15:22:48.3616218Z 	at io.vertx.core.http.impl.http1.Http1ClientConnection.checkLifecycle(Http1ClientConnection.java:789)
2026-07-10T15:22:48.3617947Z 	at io.vertx.core.http.impl.http1.Http1ClientConnection.handleResponseEnd(Http1ClientConnection.java:1009)
2026-07-10T15:22:48.3619592Z 	at io.vertx.core.http.impl.http1.Http1ClientConnection.handleHttpMessage(Http1ClientConnection.java:873)
2026-07-10T15:22:48.3621200Z 	at io.vertx.core.http.impl.http1.Http1ClientConnection.handleMessage(Http1ClientConnection.java:831)
2026-07-10T15:22:48.3622567Z 	at io.vertx.core.net.impl.VertxConnection.read(VertxConnection.java:309)
2026-07-10T15:22:48.3623662Z 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:170)
...
```

_Originally posted by @geoand in https://github.com/quarkusio/quarkus/issues/55169#issuecomment-4942522749_

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/stress/DrainTest.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/stress/DrainTest.java
@@ -74,8 +74,8 @@ public class DrainTest {
     private void runTest(String path, HttpClientOptions options) throws Exception {
         int num = 10_000;
         Vertx vertx = Vertx.vertx();
+        HttpClient client = vertx.createHttpClient(options, new PoolOptions().setHttp1MaxSize(64));
         try {
-            HttpClient client = vertx.createHttpClient(options, new PoolOptions().setHttp1MaxSize(64));
             CountDownLatch latch = new CountDownLatch(num);
             AtomicLong sum = new AtomicLong();
             AtomicReference<Throwable> failure = new AtomicReference<>();
@@ -89,7 +89,9 @@ public class DrainTest {
                             latch.countDown();
                         })
                         .onFailure(err -> {
-                            failure.compareAndSet(null, err);
+                            if (!failure.compareAndSet(null, err)) {
+                                failure.get().addSuppressed(err);
+                            }
                             latch.countDown();
                         });
             }
@@ -100,7 +102,8 @@ public class DrainTest {
             }
             Assertions.assertThat(sum.get()).isEqualTo(1_000_000_000L);
         } finally {
-            vertx.close().toCompletionStage().toCompletableFuture().get();
+            client.shutdown().await();
+            vertx.close().await();
         }
     }
 

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/stress/DrainTest.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/stress/DrainTest.java
@@ -1,9 +1,12 @@
 package io.quarkus.resteasy.reactive.server.test.stress;
 
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+
 import java.io.IOException;
 import java.io.OutputStream;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Type;
+import java.time.Duration;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
@@ -23,7 +26,6 @@ import org.assertj.core.api.Assertions;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.test.QuarkusExtensionTest;
@@ -39,6 +41,8 @@ import io.vertx.core.http.PoolOptions;
 
 public class DrainTest {
 
+    private static final long RUN_TIMEOUT = System.getenv("CI") != null ? 120 : 30;
+
     @RegisterExtension
     static QuarkusExtensionTest test = new QuarkusExtensionTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
@@ -48,27 +52,31 @@ public class DrainTest {
             .overrideConfigKey("quarkus.http.ssl.certificate.key-store-password", "secret");
 
     @Test
-    @Timeout(value = 30, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testAsyncHttp1() throws Exception {
-        runTest("/test/bytesAsync", createHttp1Options());
+        assertTimeoutPreemptively(Duration.ofSeconds(RUN_TIMEOUT), () -> {
+            runTest("/test/bytesAsync", createHttp1Options());
+        });
     }
 
     @Test
-    @Timeout(value = 30, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testSyncHttp1() throws Exception {
-        runTest("/test/bytesSync", createHttp1Options());
+        assertTimeoutPreemptively(Duration.ofSeconds(RUN_TIMEOUT), () -> {
+            runTest("/test/bytesSync", createHttp1Options());
+        });
     }
 
     @Test
-    @Timeout(value = 30, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testAsyncHttp2() throws Exception {
-        runTest("/test/bytesAsync", createHttp2Options());
+        assertTimeoutPreemptively(Duration.ofSeconds(RUN_TIMEOUT), () -> {
+            runTest("/test/bytesAsync", createHttp2Options());
+        });
     }
 
     @Test
-    @Timeout(value = 30, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testSyncHttp2() throws Exception {
-        runTest("/test/bytesSync", createHttp2Options());
+        assertTimeoutPreemptively(Duration.ofSeconds(RUN_TIMEOUT), () -> {
+            runTest("/test/bytesSync", createHttp2Options());
+        });
     }
 
     private void runTest(String path, HttpClientOptions options) throws Exception {
@@ -96,7 +104,7 @@ public class DrainTest {
                         });
             }
 
-            Assertions.assertThat(latch.await(30, TimeUnit.SECONDS)).isTrue();
+            Assertions.assertThat(latch.await(RUN_TIMEOUT, TimeUnit.SECONDS)).isTrue();
             if (failure.get() != null) {
                 Assertions.fail("Request failed", failure.get());
             }

